xidmap/xidmap.go

```diff
@@ -251,9 +237,29 @@ func (m *XidMap) AllocateUid() uint64 {
 
 // Flush must be called if DB is provided to XidMap.
 func (m *XidMap) Flush() error {
+	glog.Infof("Writing xid map to DB")
+	defer func() {
+		glog.Infof("Finished writing xid map to DB")
+	}()
+
 	for _, shard := range m.shards {
+		var err error
+		if m.writer != nil {
+			shard.Lock()
+			err = shard.trie.Iterate(func(key string, uid uint64) error {
```

Comment:
We are not doing error checking? `err` may be overwritten to `nil` by further entries?

Reply:
My bad.